doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -595,90 +584,85 @@ the conversion in hypotheses :n:`{+ @ident}`.
+   constant :g:`plus' := plus` may be unfolded and reused in
+   recursive calls, but a constant such as :g:`succ := plus (S O)` is
+   never unfolded.
 
-   Notice that only transparent constants whose name can be reused in the
-   recursive calls are possibly unfolded by :tacn:`simpl`. For instance a
-   constant defined by :g:`plus' := plus` is possibly unfolded and reused in
-   the recursive calls, but a constant such as :g:`succ := plus (S O)` is
-   never unfolded. This is the main difference between :tacn:`simpl` and :tacn:`cbn`.
-   The tactic :tacn:`cbn` reduces whenever it will be able to reuse it or not:
-   :g:`succ t` is reduced to :g:`S t`.
+   :n:`{| @reference_occs | @pattern_occs }` can limit the application of :tacn:`simpl`
+   to:
 
-.. tacv:: cbn [ {+ @qualid} ]
-          cbn - [ {+ @qualid} ]
+   - the applicative subterms whose head occurrence is the unfoldable
+     constant :n:`@qualid` or is the constant used
+     in the notation :n:`@string` (see :n:`@reference`)
+   - subterms matching a pattern :n:`@one_term`
 
-   These are respectively synonyms of :n:`cbn beta delta [ {+ @qualid} ] iota zeta`
-   and :n:`cbn beta delta - [ {+ @qualid} ] iota zeta` (see :tacn:`cbn`).
+   The behavior can be tuned using the :cmd:`Arguments` command.
 
-.. tacv:: simpl @pattern
+.. tacn:: cbn {? @reductions } {? @occurrences }
 
-   This applies :tacn:`simpl` only to the subterms matching
-   :n:`@pattern` in the current goal.
+   :tacn:`cbn` tactic was intended to be a more principled, faster and more
+   predictable replacement for :tacn:`simpl`.
 
-.. tacv:: simpl @pattern at {+ @natural}
+   :tacn:`cbn` tactic accepts the same reductions as :tacn:`cbv` and
+   :tacn:`lazy`. The behavior can be tuned using the :cmd:`Arguments` command.
 
-   This applies :tacn:`simpl` only to the :n:`{+ @natural}` occurrences of the subterms
-   matching :n:`@pattern` in the current goal.
+   The main difference between :tacn:`cbn` and :tacn:`simpl` is that
+   :tacn:`cbn` may unfold constants even when they cannot be reused in recursive calls:
+   :g:`succ t` is reduced to :g:`S t`.
 
    .. exn:: Too few occurrences.
       :undocumented:
 
-.. tacv:: simpl @qualid
-          simpl @string
+   .. flag:: Debug RAKAM
 
-   This applies :tacn:`simpl` only to the applicative subterms whose head occurrence
-   is the unfoldable constant :n:`@qualid` (the constant can be referred to by
-   its notation using :n:`@string` if such a notation exists).
+      This flag makes :tacn:`cbn` print various debugging information.
+      ``RAKAM`` is the Refolding Algebraic Krivine Abstract Machine.
 
-.. tacv:: simpl @qualid at {+ @natural}
-          simpl @string at {+ @natural}
+.. tacn:: unfold {+, @reference_occs } {? @occurrences }
 
-   This applies :tacn:`simpl` only to the :n:`{+ @natural}` applicative subterms whose
-   head occurrence is :n:`@qualid` (or :n:`@string`).
+   .. todo PR: how does this work with multiple reference_occs and occurrence #s?
```

Comment:
In the same way as all other tactics?  What's the confusion here?

Reply:
It has its this corner case: `unfold Nat.add, Nat.add in |- * at 2.` gives the incorrectly worded error `"at" clause not supported in presence of an occurrence clause.` while `unfold Nat.add in |- * at 2.` works just fine.

Reply:
Oh, geeze.  Does `unfold Nat.add in H at 1|- * at 2` work?  And is `unfold` unique here or does it work the same in tactics taking both occurrences and reference_occs

Reply:
Your unfold example works.  Also works for pattern, vm_compute, native_compute.  Didn't check the others.  I see a pattern forming :-)